02-hello-caqti-blocking/lib/repo/init.ml

```diff
@@ -0,0 +1,46 @@
+(*
+   "Unpack" the `Caqti_blocking.CONNECTION` into a module, by creating an anonymous
+    `first-class module` with the `val` keyword (on the left).
+
+   See: https://dev.realworldocaml.org/first-class-modules.html
+
+   Caqti requires this interface for further work.
+  *)
+let module_of_conn (conn : Caqti_blocking.connection) =
+  (module (val conn) : Caqti_blocking.CONNECTION)
```

Comment:
I missed this on first reading, but this is really just the identify function (with a type constraint), since `(val ...)` unpacks the module and `(module ...)` re-packs it.  I therefore think it will only lead to confusion about how first-class modules work.  My advice to the reader would be to pass around the connection as a value and only unpack it at the last moment in the local scope where it is used, e.g. by accepting a `(module Conn : ...)` pattern as a formal argument `(or` by unpacking it with a local `let module : ... = (val Conn)`).